### PR TITLE
TAXII2 Default API Root

### PR DESCRIPTION
**Pull request by BEAdi**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes:[ link to the issue](https://jira-hq.paloaltonetworks.local/browse/XSUP-14603)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py [hunk 1]
```diff
@@ -229,22 +232,37 @@ def init_roots(self):
...
+            self.api_root = roots_to_api.get(self.default_api_root)  # type: ignore[union-attr, attr-defined]
+
+        elif self.server.default:  # type: ignore[union-attr, attr-defined]
+            self.api_root = self.server.default  # type: ignore[union-attr, attr-defined]
+
+        else:
+            self.api_root = self.server.api_roots[0]  # type: ignore[union-attr, attr-defined]
+
+        # override _conn - api_root isn't initialized with the right _conn
+        self.api_root._conn = self._conn  # type: ignore[union-attr, attr-defined]
```
Comment (DeanArbel) [inline Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py:264]: Is it possible to deal with these types of errors instead of ignoring them?

**Review by DeanArbel — APPROVED** [review]
Looks good, but please see if you can better handle the type erros.

File: Packs/FeedTAXII/Integrations/FeedTAXII2/FeedTAXII2.yml [hunk 2]
```diff
@@ -177,6 +177,11 @@ configuration:
   name: feedTags
   required: false
   type: 0
+- additionalinfo: The Default API Root to use (e.g. default, public).
+  display: Default API Root to use
```
Comment (DeanArbel) [inline Packs/FeedTAXII/Integrations/FeedTAXII2/FeedTAXII2.yml:181]: What happens if we leave this empty?

Comment (BEAdi) [reply Packs/FeedTAXII/Integrations/FeedTAXII2/FeedTAXII2.yml:181]: Updated additional info

File: Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py (hunk 1 above)
Comment (BEAdi) [reply Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py]: Updated and limited ignores to the only ones needed.

File: Packs/FeedTAXII/Integrations/FeedTAXII2/FeedTAXII2.yml [hunk 3]
```diff
@@ -177,6 +177,11 @@ configuration:
   name: feedTags
   required: false
   type: 0
+- additionalinfo: The Default API Root to use (e.g. default, public). When remained empty, the server default API root is used, or the first API root if no default is defined.
```
Comment (DeanArbel) [inline Packs/FeedTAXII/Integrations/FeedTAXII2/FeedTAXII2.yml:180]: Please make sure to add this to the readme.md.
```suggestion
- additionalinfo: The Default API Root to use (e.g. default, public). If left empty, the server default API root will be used. When the server has no default root, the first available API root will be used.
```

**Pull request merged**